pkg/provider/kubernetes/kubernetes.go

```diff
@@ -94,6 +86,12 @@ func (k *ProviderKubernetes) NewClient(ctx context.Context, store esv1beta1.Gene
 		storeKind: store.GetObjectKind().GroupVersionKind().Kind,
 	}
 
+	// allow SecretStore controller validation to pass
+	// when using referent namespace.
+	if bStore.storeKind == esv1beta1.ClusterSecretStoreKind && bStore.namespace == "" {
+		return p, nil
+	}
+
```

Comment:
If Validate() is changed to allow non-referent ClusterSecretStores, we need to include a `isReferent` call here.